docs/standard/analyzers/platform-compat-analyzer.md

```diff
@@ -0,0 +1,366 @@
+  [SupportedOSPlatform("windows8.0")]
+  public void StartedWindowsSupportFromVersion8() { }
+
+  // an API was unsupported on Windows until version8.0.
+  // Then the API is removed (unsupported) from version 10.0.19041.0.
+  // The API is considered supported everywhere else without constraints.
+  [UnsupportedOSPlatform("windows")]
+  [SupportedOSPlatform("windows8.0")]
+  [UnsupportedOSPlatform("windows10.0.19041.0")]
+  public void StartedWindowsSupportFrom8UnsupportedFrom10() { }
+
+  public void Caller2()
+  {
+      DoesNotWorkOnAndroid(); // warns 'DoesNotWorkOnAndroid' is unsupported on 'android'
+
+      // warns:'StartedWindowsSupportFromVersion8' is unsupported on 'windows'  
+      // warns:'StartedWindowsSupportFromVersion8' is supported on 'windows' 8.0 and later
+      StartedWindowsSupportFromVersion8();
+
+      // warns:'StartedWindowsSupportFrom8UnsupportedFrom10' is unsupported on 'windows'  
+      // warns:'StartedWindowsSupportFrom8UnsupportedFrom10' is supported on 'windows' 8.0 and later
+      // even there were 3 diagnostics found analyzer warn only for the first 2.
+      StartedWindowsSupportFrom8UnsupportedFrom10();
+  }
+  ```
+
+## Handle reported warnings
+
+The recommended way to deal with these diagnostics is to make sure you only call platform-specific APIs when running on an appropriate platform. Following are the options you can use to address the warnings; choose whichever is most appropriate for your situation:
+
+- **Guard the call**. You can either achieve this by excluding the code at build time using `#if` and multi-targeting or by conditionally calling the code at run time. Check whether you’re running on a desired `Platform` by using one of platform-check methods, for example, `OperatingSystem.Is<Platform>()` or `OperatingSystem.Is<Platform>VersionAtLeast(int major, int minor = 0, int build = 0, int revision = 0)`.
+
+- **Mark the call site as platform-specific**. You can also choose to mark your own APIs as being platform-specific, thus effectively just forwarding the requirements to your callers. Mark the containing method or type or the entire assembly with the same attributes as the referenced platform-dependent call. [Examples](#mark-call-site-as-platform-specific).
+
+- **Assert the call site with platform check**. If you don't want the overhead of an additional `if` statement at run time, use <xref:System.Diagnostics.Debug.Assert(System.Boolean)?displayProperty=nameWithType>. [Example](#assert-the-call-site-with-platform-check).
+
+- **Delete the code**. Generally not what you want because it means you lose fidelity when your code is used by Windows users. For cases where a cross-platform alternative exists, you’re likely better off using that over platform-specific APIs.
+
+- **Suppress the warning**. You can also simply suppress the warning, either via editor.config or `#pragma warning disable`. However, this option should be a last resort when using platform-specific APIs.
```

Comment:
I recommend using `#pragma warning disable ca1416` to promote disabling the single diagnostic.